library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -389,7 +389,7 @@ public void put(Slice key, long timeBucket, Slice value)
         sizeInBytes.getAndAdd(key.length);
         sizeInBytes.getAndAdd(Long.SIZE);
       }
-      if (timeBucket > bucketedValue.getTimeBucket()) {
+      if (timeBucket >= bucketedValue.getTimeBucket()) {
```

Comment:
Should we log a warning or throw an exception in the else of this if? I don't think we want to just quietly ignore a put call...